src/prefect/cli/execute.py

```diff
@@ -66,7 +66,14 @@ def cloud_flow():
             secrets[secret] = PrefectSecret(name=secret).run()
 
         with prefect.context(secrets=secrets, loading_flow=True):
-            flow = storage.get_flow(storage.flows[flow_data.name])
+            try:
+                flow = storage.get_flow(storage.flows[flow_data.name])
+            except TypeError:
```

Comment:
I vote to catch all exceptions types here, not just `TypeError`

Reply:
We do catch all exceptions below it. In this case of a TypeError we specifically catch and raise a different message

Reply:
Another option is checking the type of exception that was raised and then handling it in this special case where we want to raise a more informative error

Reply:
Catching all errors here and handling in the same way will be confusing because other exceptions could be raised here on flow retrieval such as downloading from an S3 bucket. If we want to be explicit about catching all pickle exceptions then this should be moved inside of each storage class's individual `get_flow` function

Reply:
That type error isn't the only error you could get from a mismatched python version. For example:

```python
In [4]: value_serialized_with_pickle_5 = b'\x80\x05\x95\t\x00\x00\x00\x00\x00\x00\x00\x8c\x05hello\x94.'

In [5]: pickle.loads(value_serialized_with_pickle_5)  # load in python 3.7, only pickle <= 4 supported
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-218e273bf89d> in <module>
----> 1 pickle.loads(value_serialized_with_pickle_5)

ValueError: unsupported pickle protocol: 5
```

I think we either want to provide a generic error, or move the handling of this closer to the `pickle.loads` call. Catching a `TypeError` here feels wrong.